airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/async_job.py

```diff
@@ -0,0 +1,277 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import copy
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, List, Mapping, Optional
+
+import pendulum
+from facebook_business.adobjects.adreportrun import AdReportRun
+from facebook_business.adobjects.campaign import Campaign
+from facebook_business.adobjects.objectparser import ObjectParser
+from facebook_business.api import FacebookAdsApiBatch, FacebookResponse
+
+logger = logging.getLogger("airbyte")
+
+
+def chunks(data, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(data), n):
+        yield data[i : i + n]
+
+
+class Status(str, Enum):
+    """Async job statuses"""
+
+    COMPLETED = "Job Completed"
+    FAILED = "Job Failed"
+    SKIPPED = "Job Skipped"
+    STARTED = "Job Started"
+    RUNNING = "Job Running"
+    NOT_STARTED = "Job Not Started"
+
+
+class AsyncJob(ABC):
+    """Abstract AsyncJob base class"""
+
+    @abstractmethod
+    def start(self):
+        """Start remote job"""
+
+    @abstractmethod
+    def restart(self):
+        """Restart failed job"""
+
+    @property
+    @abstractmethod
+    def restart_number(self):
+        """Number of restarts"""
```

Comment:
is this expected to start at 0 if no restarts were created eg job succeeded on first try? might be better to name it attempt_number just to make it clear

Reply:
it is expected to be 0, I see both options possible, but will change to `attempt_number`